File: tasks/implementations/node/tasks/prime_sieve.js
const LIMITS = { s: 50000, m: 125000, l: 250000 };
module.exports = function(size) {
  const n = LIMITS[size];
  const sieve = new Array(n + 1).fill(true);
  sieve[0] = sieve[1] = false;
  for (let p = 2; p * p <= n; p++) {
    if (sieve[p]) {
      for (let i = p * p; i <= n; i += p) sieve[i] = false;
    }
  }
  let count = 0;
  for (const entry of sieve) if (entry) count++;
  console.log(count);
};

File: tasks/implementations/java/tasks/prime_sieve.java
class PrimeSieve {
    public static void run(String size, String fixturesRoot) {
        int n = switch (size) { case "s" -> 50000; case "m" -> 125000; default -> 250000; };
        boolean[] sieve = new boolean[n + 1];
        java.util.Arrays.fill(sieve, true);
        sieve[0] = false; sieve[1] = false;
        for (int p = 2; p * p <= n; p++) {
            if (sieve[p]) {
                for (int i = p * p; i <= n; i += p) sieve[i] = false;
            }
        }
        int count = 0;
        for (boolean prime : sieve) if (prime) count++;
        System.out.println(count);
    }
}
